### PR TITLE
[memory] extract MemoryData class from memory_pool to another file

### PR DESCRIPTION
**Pull request by lhs8928**
- Extract MemoryData class from memory_pool.h and make memory_data.h file
 - Add doxygen

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/memory_data.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
+
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Hyeonseok Lee <hs89.lee@samsung.com>
+ *
+ * @file   cache_pool.h
+ * @date   14 Oct 2022
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Hyeonseok Lee <hs89.lee@samsung.com>
```
Comment (jijoongmoon) [inline nntrainer/tensor/memory_data.h:9]: Plz, re-consider the author.

File: nntrainer/tensor/memory_data.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
+
```
Comment (jihochu) [inline nntrainer/tensor/memory_data.h:1]: uncessary empty line

Comment (lhs8928) [reply nntrainer/tensor/memory_data.h]: Removed

**Review by jihochu — APPROVED** [review]
LGTM!!

**Review by jijoongmoon — COMMENTED** [review]
LGTM

**Pull request merged**